metabase/metabase
# optimize loading questions only for active tab

Part of epic https://github.com/metabase/metabase/issues/29502

### Description

Previously dashboards would request the query and metadata for all dashcards, even the ones not on the current tab. This PR changes the behavior so the requests will only be sent for the cards on the current tab.


#### BE changes
The only change is to make sure all dash cards generated for magic dashboard has a `dashboard_tab_id` field like a normal dashboard do.

### How to verify

1. Create a dashboard with at least one card on multiple different tabs.
2. Open browser network tab and filter requests by "query".
3. When loading the dashboard initially and selecting tabs, only the queries for that tab's cards should be requested.

### Demo


https://github.com/metabase/metabase/assets/37751258/f0380d81-6d4f-40d5-9165-6070130aef19



### Checklist
